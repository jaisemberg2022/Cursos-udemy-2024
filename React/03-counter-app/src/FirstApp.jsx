import  PropTypes from "prop-types";
import Typography from '@mui/material/Typography';



// const newMessage = {
//     mensaje:'Hola',
//     nombre:'jaider',
// };

// const getSuma = (a,b) => {
//     return a+b;
// }

export const FirstApp = ({tittle,subtitle}) => {

  return (
    <>
        <h1 data-testId="testId">{tittle}</h1>
        <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
    tittle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    tittle:'no hay titulo',
    subtitle:'no hay subtitulo'
}

