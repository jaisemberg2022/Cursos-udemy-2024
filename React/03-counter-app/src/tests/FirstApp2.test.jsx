import {render,screen} from '@testing-library/react'
import {FirstApp} from '../FirstApp'
import { teal } from '@mui/material/colors';


describe('pruebas en <FirstApp>', () => { 
    const tittle = 'Hola soy goku';
    const subtittle ='subtitulo'

    test('Debe hacer match con snapshot', () => { 
        const {container} = render(
            <FirstApp tittle={tittle}/>
        )
        expect(container).toMatchSnapshot();
    });
    
    test('Debe mostrar el mensaje hola soy goku',()=>{
        render(<FirstApp tittle={tittle}/>)
        expect(screen.getByText(tittle)).toBeTruthy();       
    });

    test('Debe de mostrar el titulo en un h1',()=>{
        render(<FirstApp tittle={tittle}/>)
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(tittle);
    });

    test('Debe de mostrar el subtitulo enviado por props',()=>{
        render(<FirstApp tittle={tittle} subtitle={subtittle}/>)
        expect(screen.getAllByText(tittle)).toBeTruthy();       
        
    }); 

 })