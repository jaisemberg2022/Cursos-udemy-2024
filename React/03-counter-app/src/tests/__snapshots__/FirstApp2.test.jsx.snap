// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp> Debe hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="testId"
  >
    Hola soy goku
  </h1>
  <p>
    no hay subtitulo
  </p>
</div>
`;
